doc/specs/stdlib_hash_procedures.md

```diff
@@ -0,0 +1,1732 @@
+|            | Bytes     |            |          |
+|------------|-----------|------------|----------|
+|     FNV-1  |       1   |    1048576 |  0.02981 |
+|     FNV-1  |       2   |     524288 |  0.02697 |
+|     FNV-1  |       4   |     262144 |  0.02275 |
+|     FNV-1  |       8   |     131072 |  0.02431 |
+|     FNV-1  |      16   |      65536 |  0.02158 |
+|     FNV-1  |      64   |      16384 |  0.02007 |
+|     FNV-1  |     256   |       4096 |  0.01932 |
+|     FNV-1  |    1024   |       1024 |  0.02089 |
+|    FNV-1a  |       1   |    1048576 |  0.03226 |
+|    FNV-1a  |       2   |     524288 |  0.03076 |
+|    FNV-1a  |       4   |     262144 |  0.02359 |
+|    FNV-1a  |       8   |     131072 |  0.02542 |
+|    FNV-1a  |      16   |      65536 |  0.02364 |
+|    FNV-1a  |      64   |      16384 |  0.02130 |
+|    FNV-1a  |     256   |       4096 |  0.01962 |
+|    FNV-1a  |    1024   |       1024 |  0.01966 |
+|     Pengy  |       1   |    1048576 |  0.24294 |
+|     Pengy  |       2   |     524288 |  0.12066 |
+|     Pengy  |       4   |     262144 |  0.06205 |
+|     Pengy  |       8   |     131072 |  0.03138 |
+|     Pengy  |      16   |      65536 |  0.01608 |
+|     Pengy  |      64   |      16384 |  0.00669 |
+|     Pengy  |     256   |       4096 |  0.00387 |
+|     Pengy  |    1024   |       1024 |  0.00295 |
+|    Spooky  |       1   |    1048576 |  0.11920 |
+|    Spooky  |       2   |     524288 |  0.07478 |
+|    Spooky  |       4   |     262144 |  0.03185 |
+|    Spooky  |       8   |     131072 |  0.01468 |
+|    Spooky  |      16   |      65536 |  0.01503 |
+|    Spooky  |      64   |      16384 |  0.00440 |
+|    Spooky  |     256   |       4096 |  0.00290 |
+|    Spooky  |    1024   |       1024 |  0.00177 |
+
+As the tested function will typically reside in the instruction cache
+these results do not include the costs of reloading the procedure if
+hashing is intermittent. If hashing is intermittent then that can more
+severely impact the performance of  `nmhash32`, `nmhash32x`,
+`water_hash`, `pengy_hash`, and `spooky_hash` relative to
+`fnv_1_hash` and `fnv_1a_hash`.
+
+In the `src/test/hash_functions/validation` subdirectory, the Fortran
+Standard Library implements three executables to test the validity of
+the Fortran codes against the original C and C++ codes. The three
+executables must be compiled manually using the makefile
+`Makefile.validation`, and the the compiler suite used must be
+GCC's. The first executable, `generate_key_array` is 
+based on Fortran code, and generates a random sequence of 2048
+integers of kind `INT8`, and stores that sequence in the binary file
+`key_array.bin`. The second executable, `generate_hash_arrays`, reads
+the values in `key_array.bin`, and, for each complicated hash
+procedure generates a corresponding binary file containing 2049 hash
+values generated from the values in `key_array.bin`. The third
+executsble, `hash_validity_test`, reads the binary files and for each
```

Comment:
```suggestion
executable, `hash_validity_test`, reads the binary files and for each
```

Reply:
Fixed this and other spelling error found when viewing the file with an application that does spell checking. Committed and pushed.